File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/repositories/AppDataRepository.java
package nl.acidcats.tumblrlikes.core.repositories;

import android.support.annotation.Nullable;

/**
 * Created by stephan on 29/04/2017.
 */

public interface AppDataRepository {
    String getTumblrApiKey();

    void setTumblrBlog(String tumblrBlog);

    String getTumblrBlog();

    void setPincodeHash(String pinCode);

    @Nullable
    String getPincodeHash();

    long getLatestCheckTimestamp();

    void setLatestCheckTimestamp(long time);

    long getAppStopTime();

    void setAppStopTime(long appStopTime);
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/repositories/AppDataRepository.kt
package nl.acidcats.tumblrlikes.core.repositories

/**
 * Created on 24/10/2018.
 */
interface AppDataRepository {
    fun getTumblrApiKey(): String

    fun setTumblrBlog(tumblrBlog: String)

    fun getTumblrBlog(): String?

    fun setPincodeHash(pinCode: String)

    fun getPincodeHash(): String?

    fun getLastLikeTime(): Long

    fun setLastLikeTime(time: Long)

    fun getAppStopTime(): Long

    fun setAppStopTime(appStopTime: Long)

    fun getLastCheckTime(): Long

    fun setLastCheckTime(time:Long)
}